src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -92,102 +93,119 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
+                    if (!cell) {
+                        const editor = this.vscNotebook.activeNotebookEditor;
+                        const range = editor?.selections[0];
+                        if (range) {
+                            cell = editor?.document.cellAt(range.start);
+                        }
                     }
-                }
 
-                if (!cell) {
-                    return;
-                }
+                    if (!cell) {
+                        return;
+                    }
 
-                const controller = this.notebookToRunByLineController.get(cell.notebook);
-                if (controller && controller.debugCell.document.uri.toString() === cell.document.uri.toString()) {
-                    controller.continue();
-                }
-            }),
+                    const controller = this.notebookToRunByLineController.get(cell.notebook);
+                    if (controller && controller.debugCell.document.uri.toString() === cell.document.uri.toString()) {
+                        controller.continue();
+                    }
+                }, 500)
+            ),
 
-            this.commandManager.registerCommand(DSCommands.RunByLineStop, () => {
-                const editor = this.vscNotebook.activeNotebookEditor;
-                if (editor) {
-                    const controller = this.notebookToRunByLineController.get(editor.document);
-                    if (controller) {
-                        sendTelemetryEvent(DebuggingTelemetry.endedSession, undefined, { reason: 'withKeybinding' });
-                        controller.stop();
+            this.commandManager.registerCommand(
+                DSCommands.RunByLineStop,
+                throttle(() => {
+                    const editor = this.vscNotebook.activeNotebookEditor;
+                    if (editor) {
+                        const controller = this.notebookToRunByLineController.get(editor.document);
+                        if (controller) {
+                            sendTelemetryEvent(DebuggingTelemetry.endedSession, undefined, {
+                                reason: 'withKeybinding'
+                            });
+                            controller.stop();
+                        }
                     }
-                }
-            }),
+                }, 1000)
+            ),
 
-            this.commandManager.registerCommand(DSCommands.RunAndDebugCell, async (cell: NotebookCell | undefined) => {
-                sendTelemetryEvent(DebuggingTelemetry.clickedRunAndDebugCell);
-                const editor = this.vscNotebook.activeNotebookEditor;
-                if (!cell) {
-                    const range = editor?.selections[0];
-                    if (range) {
-                        cell = editor?.document.cellAt(range.start);
+            this.commandManager.registerCommand(
+                DSCommands.RunAndDebugCell,
+                throttle(async (cell: NotebookCell | undefined) => {
```

Comment:
I worry it will feel broken if this command sometimes doesn't work if stuff happens too quickly

Reply:
Yeah throttle of 1 second seems like way too much. Maybe like 100ms